src/java.base/share/classes/java/util/Random.java

```diff
@@ -150,6 +308,23 @@ private static long initialScramble(long seed) {
         return (seed ^ multiplier) & mask;
     }
 
+    /**
+     * Returns an instance of {@code Random} that delegates method calls to the {@link RandomGenerator}
+     * argument. If the generator is an instance of {@code Random}, it is returned. Otherwise, this method
+     * returns an instance of {@code Random} that delegates all methods except {@code setSeed} to the generator.
+     * The returned instance's {@code setSeed} method always throws {@link UnsupportedOperationException}.
+     * The returned instance is not serializable.
+     *
+     * @param generator the {@code RandomGenerator} calls are delegated to
+     * @return the delegating {@code Random} instance
+     */
+    public static Random from(RandomGenerator generator) {
+        if (generator instanceof Random rand)
+            return rand;
+
+        return new RandomWrapper(generator);
+    }
+
```

Comment:
I'd suggest adding

    Objects.requireNonNull(generator)

as the first line of this method. In fact if null is passed, NPE is thrown already, as the `instanceof` will be false and then the `RandomWrapper` constructor will throw NPE. However, verifying this requires a bit of hunting around and some flow analysis to determine this. Might as well make the null check explicit at the top of this method. The now-redundant check can be removed from `RandomWrapper`.